src/util/ThreadSafeQueue.h

```diff
@@ -16,11 +16,39 @@
 
 #include "util/Exception.h"
 #include "util/ExceptionHandling.h"
-#include "util/Generator.h"
+#include "util/Iterators.h"
 #include "util/jthread.h"
 
-namespace ad_utility::data_structures {
+namespace {
```

Comment:
As noted in the other PR, anonymous namespace in the header is an antipattern (might lead to linking problems)
use a `....:::detail `namespace for implementation details.